### ISSUE TITLE
OTP 25: Illegal return value from inet_res:resolve/3

### ISSUE THREAD

**Issue by mworrell** [erlang/otp#6015]
**Describe the bug**

`inet_res:lookup/3` returns a string when called with an argument that is an IP address or a string with an IP address.

**To Reproduce**

```erlang
inet_res:resolve("127.0.0.1", in, a).
```

Returns on OTP 25:

```erlang
"1.0.0.127.in-addr.arpa"
```

**Expected behavior**

The returned string is not according to the spec of the `resolve/3` function.

Expected is what OTP 24 returns:

```erlang
{ok,#dns_rec{header = #dns_header{id = 2,qr = true,
                                  opcode = query,aa = false,tc = false,rd = true,ra = true,
                                  pr = false,rcode = 0},
             qdlist = [#dns_query{domain = "1.0.0.127.IN-ADDR.ARPA",
                                  type = a,class = in}],
             anlist = [],nslist = [],arlist = []}}
```

**Affected versions**

OTP25

**Comment by RaimoNiskanen** [comment]
@leonardb: Absolutely correct.  I am working on a patch...

**Comment by leonardb** [comment]
@mworrell are you sure 24 returns an actual `dns_rec`?
My testing on OTP 23 and 24 show that it returns `{error, nxdomain}`

```
leonard@captmarvel:~$ asdf local erlang 24.2
leonard@captmarvel:~$ erl
Erlang/OTP 24 [erts-12.2] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1] [jit]

Eshell V12.2  (abort with ^G)
1> inet_res:resolve("127.0.0.1", in, a).  
{error,nxdomain}
2> 


leonard@captmarvel:~$ asdf local erlang 23.3.4.8
leonard@captmarvel:~$ erl
Erlang/OTP 23 [erts-11.2.2.7] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1] [hipe]

Eshell V11.2.2.7  (abort with ^G)
1> inet_res:resolve("127.0.0.1", in, a).
{error,nxdomain}
2> 

leonard@captmarvel:~$ asdf local erlang 25.0
leonard@captmarvel:~$ erl
Erlang/OTP 25 [erts-13.0] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1] [jit:ns]

Eshell V13.0  (abort with ^G)
1> inet_res:resolve("127.0.0.1", in, a).
"1.0.0.127.in-addr.arpa"
2> 

```

**Comment by mworrell** [comment]
Yes, guess macOS has some entries for 127.0.0.1
But that is besides the point of this issue.

The function `inet_res:resolve/3` should _never_ return just a string.  Which it does currently on OTP 25.

**Comment by RaimoNiskanen** [comment]
I would like to take the opportunity to advocate testing of our Release Candidates.

If I figured this out right, the bug must have been published in OTP-25.0-rc1 (and our test cases did not cover it, which is another problem).

**Comment by mworrell** [comment]
👍  Will try running our tests through the RC releases when they are available on kerl.

**Comment by RaimoNiskanen** [comment]
Fixed in PR #6020.
Thank you for your bug report!

**Comment by mworrell** [comment]
@RaimoNiskanen Thank you for fixing this so quickly!
